pkg/provider/azure_loadbalancer.go

```diff
@@ -1928,33 +1928,85 @@ func (az *Cloud) buildHealthProbeRulesForPort(annotations map[string]string, por
 
 	properties := &network.ProbePropertiesFormat{}
 	var err error
-	if port.AppProtocol == nil {
-		if port.AppProtocol, err = consts.GetAttributeValueInSvcAnnotation(annotations, consts.ServiceAnnotationLoadBalancerHealthProbeProtocol); err != nil {
+
+	// order - Specific Override
+	// port_ annotation
+	// global annotation
+
+	// Select Protocol
+	//
+	var protocol *string
+
+	// 1. Look up port-specific override
+	protocol, err = consts.GetHealthProbeConfigOfPortFromK8sSvcAnnotation(annotations, port.Port, consts.HealthProbeParamsProtocol)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse annotation %s: %w", consts.BuildHealthProbeAnnotationKeyForPort(port.Port, consts.HealthProbeParamsProtocol), err)
+	}
+
+	// 2. If not specified, look up from AppProtocol
+	// Note - this order is to remain compatible with previous versions
+	if protocol == nil {
+		protocol = port.AppProtocol
+	}
+
+	// 3. If protocol is still nil, check the global annotation
```

Comment:
@MartinForReal I don't think this order makes sense. I would imagine the order we want is:

1. Local Port Annotation
2. Global Annotation
3. appProtocol in port
4. Default of TCP

However it seems that would be a breaking change. In the previous code it only applies the global annotation if appProtocol is not set. We want to set appProtocol for lots of reasons not just health checking.

Ideally it would be:
1. per-port annotation
2. global annotation
3. appProtocol
4. default (TCP)

If we agree to make the breaking change then I'll make a second PR to allow us to globally set the health probe protocol and probe port... Otherwise its fine to leave this as is.

Reply:
IMHO, I think the proper order would be:
Local Port Annotation
appProtocol in port
Global Annotation (option is removed from doc and code is not remove for backward compatibility)
Default of TCP

Reply:
That sounds good to me especially as the global annotation is deprecated :)